model/workflow_run_event_context_test.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWorkflowRunEventContext(t *testing.T) {
+	t.Run("Test PR", func(t *testing.T) {
+		context := newWorkflowRunEventContext(createWorkflowRunEvent(t, "workflow_run_event_pr.json"))
+		assert.Equal(t, "ab7a32c308ac42df77385bbb5e97f0e3aac5c42f", context.GetCommitHash())
+		assert.Equal(t, "", context.GetConclusion())
+		assert.Equal(t, "workflow_run", context.GetEvent())
+		assert.Equal(t, int64(1854), context.GetInstallationID())
+		assert.Equal(t, "feat/cld-3876-create-github-release-bot-for-unified-ci", context.GetName())
+		assert.Equal(t, "mattermost/release-bot", context.GetRepository())
+		assert.Equal(t, "queued", context.GetStatus())
+		assert.Equal(t, "pr", context.GetType())
+		assert.Equal(t, "Build", context.GetWorkflow())
+		assert.Equal(t, int64(2926155304), context.GetWorkflowRunID())
+		assert.Equal(t, false, context.IsFork())
+	})
+	t.Run("Test Branch", func(t *testing.T) {
+		context := newWorkflowRunEventContext(createWorkflowRunEvent(t, "workflow_run_event_branch.json"))
+		assert.Equal(t, "ab7a32c308ac42df77385bbb5e97f0e3aac5c42f", context.GetCommitHash())
+		assert.Equal(t, "completed", context.GetConclusion())
+		assert.Equal(t, "workflow_run", context.GetEvent())
+		assert.Equal(t, int64(1854), context.GetInstallationID())
+		assert.Equal(t, "feat/cld-3876-create-github-release-bot-for-unified-ci", context.GetName())
+		assert.Equal(t, "mattermost/release-bot", context.GetRepository())
+		assert.Equal(t, "finished", context.GetStatus())
+		assert.Equal(t, "branch", context.GetType())
+		assert.Equal(t, "Build", context.GetWorkflow())
+		assert.Equal(t, int64(2926155304), context.GetWorkflowRunID())
+		assert.Equal(t, false, context.IsFork())
+	})
+	t.Run("Test Tag", func(t *testing.T) {
+		context := newWorkflowRunEventContext(createWorkflowRunEvent(t, "workflow_run_event_tag.json"))
+		assert.Equal(t, "ab7a32c308ac42df77385bbb5e97f0e3aac5c42f", context.GetCommitHash())
+		assert.Equal(t, "completed", context.GetConclusion())
+		assert.Equal(t, "workflow_run", context.GetEvent())
+		assert.Equal(t, int64(1854), context.GetInstallationID())
+		assert.Equal(t, "", context.GetName())
```

Comment:
same here

Reply:
Again the sample json data has no information about the name. So for the null data they converted to empty string.